osgbuild/importer.py

```diff
@@ -399,10 +400,18 @@ def move_to_cache(srpm, upstream_root):
     return dest_file
 
 
-def main(argv=None):
-    if argv is None:
-        argv = sys.argv
+def get_sha1sum(file_path):
+    """Return the SHA1 checksum of the file located at `file_path` as a string."""
+    out, ret = utils.sbacktick(["sha1sum", file_path])
+    if ret != 0:
+        raise Error("Unable to get sha1sum of %s: exit %d when running sha1sum" % (file_path, ret))
+    match = re.match(r"[a-f0-9]{40}", out)
+    if not match:
+        raise Error("Unable to get sha1sum of %s: unexpected output: %s" % (file_path, out))
+    return match.group(0)
```

Comment:
How would you feel about using [`fetch_sources.download_uri()`](https://github.com/opensciencegrid/osg-build/blob/7c52ca1a107dc12a96663be655b1e52877d866fb/osgbuild/fetch_sources.py#L119-L134)?

It downloads (without slurping the whole thing into memory at once) and returns the `sha1sum` value.

Reply:
I'll give it a shot.